include/Pothos/Object/RegisterToString.hpp

```diff
@@ -0,0 +1,102 @@
+///
+/// \file Object/RegisterToString.hpp
+///
+/// A convenience function to register functions to be used with
+/// Object::toString().
+///
+/// \copyright
+/// Copyright (c) 2020 Nicholas Corgan
+/// SPDX-License-Identifier: BSL-1.0
+///
+
+#pragma once
+
+#include <Pothos/Callable.hpp>
+#include <Pothos/Exception.hpp>
+#include <Pothos/Plugin.hpp>
+#include <Pothos/Util/Templates.hpp>
+#include <Pothos/Util/TypeInfo.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+
+namespace Pothos
+{
+
+template <typename T>
+static void registerToStringFunc(
+    const std::string& typeName,
+    const Pothos::Callable& constRefCallable,
```

Comment:
so rather than passing `Pothos::Callable(&pothosBlockToString),` for example, just pass `&pothosBlockToString` and make this parameter a `std::function<std::string(const T &)>`, that way the compiler can check the arg and return types rather than the runtime checks below with the InvalidArgumentException.

Also the pointer wrappers can directly call the same function without the extra registry indirection, so

`return "Pointer: " + Pothos::Object(std::ref(*ptr)).toString();` becomes `return "Pointer: " + fmt_func(*ptr);`

Reply:
Done.